### PR TITLE
Issue #Sb-6150 fix: bug fix and test case

### PR DESCRIPTION
**Pull request by revathitarento**

### CODE REVIEW

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts [hunk 1]
```diff
@@ -228,6 +229,13 @@ export class ProminentFilterComponent implements OnInit, OnDestroy {
   isObject(val) { return typeof val === 'object'; }
 
   applyFilters() {
+    const compareFormInputQueryparams = _.isEqual(this.formInputData,  this.queryParams);
+    if (compareFormInputQueryparams) {
+      this.isFiltered = true;
+    }  else {
+      this.isFiltered = false;
+    }
+    if (this.isFiltered === false) {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts]: code is not formatted

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts [hunk 2]
```diff
@@ -228,6 +229,13 @@ export class ProminentFilterComponent implements OnInit, OnDestroy {
   isObject(val) { return typeof val === 'object'; }
 
   applyFilters() {
+    const compareFormInputQueryparams = _.isEqual(this.formInputData,  this.queryParams);
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts]: do we need the constant as it is only used in a if

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts [hunk 3]
```diff
@@ -228,6 +229,13 @@ export class ProminentFilterComponent implements OnInit, OnDestroy {
...
+    if (compareFormInputQueryparams) {
+      this.isFiltered = true;
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts]: cant we make the isFiltered = false as default and if compareFormInputQueryparams make it true

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.spec.ts [hunk 4]
```diff
@@ -68,6 +68,19 @@ describe('ProminentFilterComponent', () => {
     component.applyFilters();
     expect(router.navigate).toHaveBeenCalledWith([undefined], { queryParams: queryParams });
   });
+
+  it('Filter results should not change on call of  apply filters for same params', () => {
+    const router = TestBed.get(Router);
+    component.formInputData = {
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.spec.ts]: can't we say component.formInputData = queryParams = { 'gradeLevel': ['Grade 1', 'Grade 2'], 'medium': ['English'] }

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.spec.ts [hunk 5]
```diff
@@ -68,6 +68,19 @@ describe('ProminentFilterComponent', () => {
...
+      'gradeLevel': ['Grade 1', 'Grade 2'], 'medium': ['English'],
+    };
+    const queryParams = { 'gradeLevel': ['Grade 1', 'Grade 2'], 'medium': ['English'] };
+    const compareFormInputQueryparams = _.isEqual(component.formInputData,  queryParams);
```
Comment (rajeevsathish) [inline src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.spec.ts]: what is the use of this variable compareFormInputQueryparams?

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.spec.ts (hunk 4 above)
Comment (revathitarento) [reply src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.spec.ts]: it is throwing error

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts (hunk 1 above)
Comment (revathitarento) [reply src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts]: done

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts (hunk 2 above)
Comment (revathitarento) [reply src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts]: removed assigning to a constant

File: src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts (hunk 3 above)
Comment (revathitarento) [reply src/app/client/src/app/modules/core/components/prominent-filter/prominent-filter.component.ts]: yes, its false when declared

**Pull request merged**